core/types/set_code_tx.go

```diff
@@ -245,11 +245,13 @@ func (tx *SetCodeTransaction) DecodeRLP(s *rlp.Stream) error {
 	if b, err = s.Bytes(); err != nil {
 		return err
 	}
-	if len(b) != 20 {
+	if len(b) > 0 && len(b) != 20 {
```

Comment:
The previous logic of requiring 20 bytes for `To` was fine because EIP-7702 transactions, similar to blob transactions, must have a non-nil recipient.